### ISSUE TITLE
Allow shields to whitelist an entire site, including subdomains

### ISSUE THREAD

**Issue by TitaniumCoder477** [brave/brave-browser#5290]
I switched from Chrome to Brave and then realized the Brave Shield has no domain whitelist feature. My work in IT requires that I use many web applications for which I do not want the shield operating. For example, I remote into dozens of servers each day using LogMeIn. Each URl is different, but the domain is always logmein.com. I would like to whitelist that domain so that each sub-domain is automatically whitelisted. AdBlockPlus has this feature. So now I have to move back to Chrome. I feel this is a basic feature that should be incorporated into the Brave Shield.

**Comment by jonathansampson** [comment]
Other users have also requested this functionality:

- https://twitter.com/nicksergeant/status/1169044157568077824

**Comment by gaui** [comment]
Any news? This is annoying as hell when I use codesandbox.io

**Comment by slickware** [comment]
Pinging this again. This is an incredibly necessary request, why has this sat here since July?

**Comment by mpurrutu** [comment]
Yeah this would be useful

**Comment by David-Martel** [comment]
This would be a very useful feature

**Comment by vistalba** [comment]
Need this feature too.

**Comment by Elenaltarien** [comment]
Also need this for using LogMeIn for work

**Comment by geeper** [comment]
This is a close to a deal breaker for me. absolute must have

**Comment by ronanyeah** [comment]
I need this to get Google OAuth working.

**Comment by Amaneusz** [comment]
Any news on this? I got plenty of apps within my company's domain and their number is growing constantly - I'd love to see this feature get through

**Comment by bsclifton** [comment]
cc: @rebron

**Comment by Luminus** [comment]
I found this issue because I was looking for the same thing. It would be great to have a setting for doing this.

**Comment by AdamSC1-ddg** [comment]
Brave currently does allow for whitelisting via the brave://adblock/ page using adblock syntax:

`@@||ads.example.com/notbanner^$~script`

or entire sites:

`@@||example.com^$document`

if the OP needs to whitelist logmein they should be able to do so (with all subdomains) by adding:

`@@||logmein.com^`  

What's missing currently is the ability to add URLs or to use a custom schema like ABP has to subscribe to a list:

`abp:subscribe?location=https%3A%2F%2Fwww.example.com%2Fwhitelist%2Fsample-whitelist.txt%26amp%3Btitle%3DExample%20Whitelist`

Brave would likely be able to solve a number of extension compatibility issues if there was a programmatic way to add exceptions or to subscribe to lists so that extensions could provide configuration files that users could decide to enable to deal with compatibility issues.

**Comment by Luminus** [comment]
Thanks @AdamSC1-ddg.

I just tried this with `@@||wordpress.com^` as a quick test to whitelist all WordPress.com sites, but the shields are still operational on them.

Is there something I'm missing? 

I've also tried the `@@||wordpress.com^$document` variant to no avail.

**Comment by geeper** [comment]
I tried the above and it didn't work for me either.

**Comment by AdamSC1-ddg** [comment]
I had managed to get this to work a while ago when targeting Outbrain ads, but, having tried again there is an issue.

To further test this I  noted that:

1. Custom filters do block. For example `||redditstatic.com^` will break Reddit.
2. Custom whitelisting will overrule custom blocks. For example, placing `@@||redditstatic.com^` one line under `||redditstatic.com^` over rules the blocking.

It seems that it isn't the case that whitelisting in general isn't working - but, rather that we are unable to have whitelisting over rule built-in lists and regional lists. I would guess based on the order they are being applied in? (cc: @rebron @bsclifton )

(Also @Luminus - the rule `@@||wordpress.com^` wouldn't turn off shields for all WordPress sites, it would prevent any request from WordPress.com from being blocked on any site.)

**Comment by Luminus** [comment]
@AdamSC1-ddg thanks for the explanation.

So in essence, it isn't quite possible just yet to whitelist a domain and all its subdomains so that shields are down for them.

Is this something that you think will get implemented anytime soon?

**Comment by AdamSC1-ddg** [comment]
> @AdamSC1-ddg thanks for the explanation.
> 
> So in essence, it isn't quite possible just yet to whitelist a domain and all its subdomains so that shields are down for them.
> 
> Is this something that you think will get implemented anytime soon?

In theory doing `@@||wordpress.com^$document` should whitelist all calls (same as disabling shields) on wordpress.com and its subdomains. Where as `@@||wordpress.com^` should whitelist all calls to WordPress from other sites.

Not sure why its failing, but, I tagged the Brave staff who were involved in that convo above so hopefully they can investigate a fix.

**Comment by bsclifton** [comment]
Multiple +1's from https://github.com/brave/brave-browser/issues/7680

Including this great list from @Brave-Matt:
> Some additional reports:
>
> - https://community.brave.com/t/shields-is-blocking-webgl-on-itch-io/81402
> - https://community.brave.com/t/how-to-widen-scope-of-saved-site-exceptions/13715
> - https://community.brave.com/t/cant-deny-all-and-whitelist-cookies-in-version-0-57-18/39166/2
> - https://community.brave.com/t/bravew-shields-league-of-legends/78761
> - https://community.brave.com/t/a-manually-edited-whitelist-of-sites/89953
> - https://community.brave.com/t/testing-brave-shields/39899/3
> - https://www.reddit.com/r/brave_browser/comments/bwt4sw/whitelist_with_wildcard/
>
> Happy to find/add more if necessary.

**Comment by bsclifton** [comment]
cc: @rebron for triage

**Comment by adamreisnz** [comment]
This is probably a reason to move back to Chrome. I really don't feel inclined to whitelist all our local IP addresses one by one...

**Comment by rebron** [comment]
@karenkliu Let's see if we can move this one along. We're looking for a list exception to turn off shields for specified domains. @AdamSC1-ddg mentioned using brave://adblock Custom Filter rules. a) It's possible we keep this as an advanced user feature and use the Custom Filter rules instead of adding UI to this. cc: @yrliou 

We could also go with using b) brave://settings/content and introduce Shield settings here to manage a list or c) add a pref to brave://settings/shields. Or something else entirely. 

@adamreisnz Thanks for the feedback and we do want to see why Brave users would possibly even consider moving back to Chrome and address those issues in timely manner. With our speed and privacy advantages https://webtest.app/ we hopefully can keep you. Those seconds add up.

**Comment by wcpines** [comment]
@rebron
> . . . It's possible we keep this as an advanced user feature and use the Custom Filter rules instead of adding UI to this . . . Or something else entirely.

I just did a test where I whitelisted `@@||api.amplitude.com^` via `chrome://adblock/`.  When I reloaded the page, the call was still listed as being blocked

![image](https://user-images.githubusercontent.com/11494834/73483419-40b2e080-4354-11ea-8e3b-c057c13e7382.png)


I would think this detailed view is a great place to add a slider toggle to whitelist a domain right from the UI, rather than having to load a separate advanced settings page.  I believe this would be similar to the Ghostery ux-- "for website x, allow 3rd party calls to domain y"

My $0.02

**Comment by DeltaMurillo** [comment]
I do not use Brave cause this. I preffer to keep using mozilla and NoScript, wich lets u whilelist sites and trackers one by one.

Having to unblock manually all the time is annoying.

**Comment by nsoui** [comment]
+1 here, too. We have a huge number of internal pages which I could whitelist with one line, if this were possible.

**Comment by mikeycrawford** [comment]
+1 

I'm trying to get out IT support over to Brave from Chrome, but this issue is preventing me recommending Brave currently.  We use LogMeIn extensively and the URL for each session changes each time we connect, e.g.

https://testpc-zmddtrpcys.app03-32.logmein.com/

and cross-site trackers enabled breaks our ability to remote on to the machine.   Yes, we can untick it each time but ideally we want to disable it for *.logmein.com.

Watching this thread with interest.

**Comment by jgregmac** [comment]
+1
I am unable to authenticate to portal.azure.com (Microsoft Azure management) because my company is integrating with Duo Security as an MFA provider.  Brave is interpreting some aspect of the cross-site MFA process as Ad or Tracking related, and thus breaking authentication.

The redirects happen so fast that I am unable to manually open the Shields UI to disable Shields on all sites involved.  Adding the sites I believe to be involved in the exchange to the brave://adblock white list (in both @@||site.com^ and @@||site.com^$document format) is ineffective.

I am falling back to use of Firefox for Azure access, but I would prefer to use Brave for everything.  I can't recommend this product to colleagues until it is compatible with all of the tools that we need to use to get our jobs done.

**Comment by frank6tg** [comment]
Definitely need this resolved. My bank recently changed its authentication when you sign in and it takes you to a URL on a subdomain that has "shields up" But I can't white list by clicking the shield because then it reloads the page / can't login to bank b/c the authentication process is interrupted.

**Comment by InsaneSplash** [comment]
+1

**Comment by andersek1** [comment]
C'mon,
Why is this still not fixed? Just add a button next to the cross-site trackers in the shield "disable". 

This is a huge issue for adotion.

**Comment by jgregmac** [comment]
Adding a little color to my previous comment.  The scope of our problem just grew from "administrative access to a cloud service management portal" to "all institutional access to email".

My employer (a major university in the northeast) currently is rolling out an MFA-everywhere initiative.  Soon, all faculty, staff, and students will need to use our external MFA provider to access all institutional web services.  At that point, it will be impossible for our constituents to use Brave for access to email, calendar, and collaboration tools.  Without an easy mechanism to unblock Office 365 and Duo Security, Brave adoption here will drop to zero.

**Comment by JStrick** [comment]
I need this as well for whitelisting my employer's sites. You'd think with nearly a year there would have been time to implement some simple version of this?

**Comment by malcolmocean** [comment]
I'll add that I want this because without it _twitter embeds don't show emoji_

Here's an image that illustrates both how many emoji are blocked by the shield (well, a third of them) as well as what the tweet looks like without it.

Personally I basically think that blocking twitter emoji should be removed from Brave altogether—if there's an embedded tweet, twitter can already track you. It's probably in the adblock since those emoji are served from syndication.twitter.com.

![image](https://user-images.githubusercontent.com/788384/83173888-8c4d7f00-a0e7-11ea-919b-a709f2d2bf9a.png)

**Comment by mcint** [comment]
Hiya away from twitter, @malcolmocean. It looks like adding `@@||syndication.twitter.com^` to the `brave://adblock/` custom blocklist would allow all requests to go through.

> the rule @@||wordpress.com^ ... would prevent any request from WordPress.com from being blocked on any site.)
https://github.com/brave/brave-browser/issues/5290#issuecomment-573074690

(Unless you need cookies on those requests).

---
Is there a way to modify cookie settings for the page without loading it? Or modify the site-specific cookies settings globally, by domain?

I'm having an issue using Brave to read papers with a library proxy. Each new journal site has a new-journal-site.libproxy.univ.edu sub-domain, sometimes multiple redirects (e.g. for the journal's auth redirect). 

It's tricky to coerce Brave to "recognize" it's on a domain so that I can update the site-shield settings. It would be sufficient (for me) to be able to update shield settings without successfully loading the domain. This might be too complicated given entanglement in Chromium network stack 😱

**Comment by rmclaughlin-nelnet** [comment]
I love Brave, but this issue is causing my work to suffer. I am now forced to switch back to Chrome so I can do my job.

**Comment by chughtai** [comment]
I second what everyone has already said. I also cannot check organisational email (hosted outlook with ADFS etc) without completely turning off the shield.

**Comment by alexi21** [comment]
Unable to log into my switch within my LAN because of this... Brave is great but without this I have to open a different browser for managing my network.

**Comment by jamiehowarth0** [comment]
I have issues with implementing privacy policies with Iubenda because of this.

**Comment by terah** [comment]
On occasion an Office 365 website will fall into a redirect loop because of this.  Today it was the forms app.

**Comment by alfonsojohan** [comment]
I have added the rule as recommended in the earlier post and have been successful in whitelisting by domain.  currently running brave version 1.18.75 on windows 10 pro.

**Comment by rawtaz** [comment]
Another example of this being a complete blocker: When using BrowserStack Local, which is a software you run locally to "tunnel" requests to local (e.g. development) websites from BrowserStack's remote browsers into your local site, the browser you're accessing BrowserStack Live (the remote browsers for manual testing) with needs to connect to `http://127.0.0.1:45691/?_=1610159247632` (number changes on every request), and when you have the shield up for live.browserstack.com (which you want, as there's a bunch of tracking there), the requests to 127.0.0.1:45691 (where the BrowserStack Local software is listening) fails with "GET http://127.0.0.1:45691/?_=1610159247632 net::ERR_BLOCKED_BY_CLIENT". Looking at the shield, these requests are also listed under trackers/ads that have been blocked.

I tried adding both `@@||127.0.0.1^` and `@@||127.0.0.1:45691^` (also tried `@@||127.0.0.1:45691^$document`) as the only line in the custom filters at `brave://adblock`, but neither of them disabled the blocking of these requests, even after reloading or restarting the tab with live.browserstack.com in it. As it stands now, for a web developer Brave is completely useless for anything but the very basic parts of web development, and I hope that a whitelist feature comes very soon. That said I totally understand and respect that I have no right to complain about this, so that's not my intent here - it's obviously up to the core developers to decide what they want to spend their time on!

**Comment by AJamesPhillips** [comment]
This feature would make it easier to use the hypothes.is plugin .  At the moment you have to login every time you visit a new page or refresh it.

**Comment by fedayn1** [comment]
https://evolvingviews.com/2020/03/how-to-setup-brave-adblock-whitelist/#:~:text=In%20the%20address%20bar%2C%20go,using%20the%20Adblock%20Plus%20syntax.

**Comment by bsclifton** [comment]
Short term, we could add a checkbox for `Use these shield settings for subdomains too` into the Advanced version of the shields UI (maybe above `Simple View`)
![image](https://user-images.githubusercontent.com/4733304/114352001-3081be00-9b20-11eb-9c9b-7a2a40f90d4d.png)

Longer term, we could expose this wildcarding via a UI where users could view/edit/add domains for shield settings (see https://github.com/brave/brave-browser/issues/10829 and https://github.com/brave/brave-browser/issues/12782).

**Comment by ashutoshsaboo** [comment]
@bsclifton this is not just a problem with subdomains. See #12782 with 302 redirects from page X to Y (say both are on different sub domains as well as domains) then how would this work out if you wanted to change shield settings for site X? The thing is that it doesn't redirect me from X to correct auth page because of shield issues (otherwise with shields turned on it 302 redirects to an error page currently), so it's really a hard requirement to change shield settings JUST for site X. 

Without exaggeration - almost everyone in my org who's wanted to use Brave has tried it, been frustrated with this, and eventually struggled and moved to either Firefox/Chrome. 

As I mentioned in #12782 as well as seeing above, a lot of folks seem to have similar requests for auth to work correctly for internal sites. 

The only way so far is to completely turn off the global shields (atleast that's what I'm only aware, feel free to point to any alternatives) - which is an extremely sub-optimal and privacy forging solution (since turning off global shields here). 

For the short term solution, why not just add shield settings to where you can currently change site specific settings like flash/camera/audio access etc for a particular site? Would that be too much effort? Before the long term is implemented, this would be a right mix IMO as it's still easy to access the page if not the most user friendly way. And will still help achieve use-case like the one i mention easily.

**Comment by jtheletter** [comment]
An option to lower shields for all subdomains would improve my experience with AWS Console.

**Comment by david0178418** [comment]
This is probably the only consistent pain point I've had with Brave.  More sites due authorization redirects, and it's a giant pain to lower global shield settings just to get the auth then turn them back on.

**Comment by jgregmac** [comment]
Recent posts here triggered me to go back and try Brave again to see if the user experience has gotten any better.

I managed to make it though about two days of nearly constant use before running into something untenable.  Interactions between Duo Security and Microsoft 365 services have improved... I now can perform essential SSO operations without needing to drop shields.  However, when trying to use the Zoom plugin to Outlook Web App I ran into blocking behavior from "Shields Up".

So we still aren't there.  I am going to discontinue use of Brave pending further developments.

**Comment by rushi** [comment]
I would really like it as well so that I can whitelist my entire company's domain. Otherwise I have to endup whitelisting every domain

**Comment by bridiver** [comment]
I am seeing some comments where people say that they are forced to go back to Chrome, but I'm having trouble understanding that because Brave's features in this area are a superset of Chrome's. There is no blocking capability in Chrome that does not exist in Brave as well. Also there's nothing stopping you from using Adblock Plus in Brave. I'm not necessarily arguing for or against this change, but I don't understand how going back to Chrome is the solution when Brave has the same controls (and more).

**Comment by david0178418** [comment]
[bridiver](https://github.com/bridiver) I think you might be missing the core issue. People are not going back to Chrome due to better blocking capabilities. Quite the opposite.

If I have a bill to pay and auth redirects traverse multiple sub domains and/or domains, then I'm stuck. I can't manually set blocking without being the browser being on that domain.  Rather than try and hack at it to get the settings adjusted or globaly turn everything off while I do what I need, it's significantly easier to use another browser for that case.

Having a way to manually adjust per-domain/sub domain without actually being on it would help address this.

That, however, would be a partial solution. We'd still have to open the devtools network tab to see what domains were hit in the flow.  Not a bit deal for those of us posting here. However, a regular user would need some other solution.

**Comment by bridiver** [comment]
> [bridiver](https://github.com/bridiver) I think you might be missing the core issue. People are not going back to Chrome due to better blocking capabilities. Quite the opposite.
> 
> If I have a bill to pay and auth redirects traverse multiple sub domains and/or domains, then I'm stuck. I can't manually set blocking without being the browser being on that domain. Rather than try and hack at it to get the settings adjusted or globaly turn everything off while I do what I need, it's significantly easier to use another browser for that case.
> 
> Having a way to manually adjust per-domain/sub domain without actually being on it would help address this.
> 
> That, however, would be a partial solution. We'd still have to open the devtools network tab to see what domains were hit in the flow. Not a bit deal for those of us posting here. However, a regular user would need some other solution.

I understand what people are wanting to do, what I'm not understanding is why you'd go back to Chrome. I'm wondering if people just don't realize that Brave also has all the same settings that Chrome does (in brave://settings/content) or if there's some other reason. Maybe they don't want to deal with abp settings plus Brave settings? Seems like throwing the baby out with the bathwater because Brave provides many other protections independently of shields, but I guess that's a personal choice.

**Comment by david0178418** [comment]
I can't speak for the others as I only switch back for the specific sites that are pain points. But I'd guess that they likely just want to use one browser. And if they have sites that simply don't work in brave, then Brave is no longer an option for them.

Even if they use some plugins with Chrome, it's easier just to temporarily turn them off globally is just a matter of clicking on the extention icon vs having to dig into the settings of brave.

**Comment by OpenOrg-gg** [comment]
> > [bridiver](https://github.com/bridiver) I think you might be missing the core issue. People are not going back to Chrome due to better blocking capabilities. Quite the opposite.
> > If I have a bill to pay and auth redirects traverse multiple sub domains and/or domains, then I'm stuck. I can't manually set blocking without being the browser being on that domain. Rather than try and hack at it to get the settings adjusted or globaly turn everything off while I do what I need, it's significantly easier to use another browser for that case.
> > Having a way to manually adjust per-domain/sub domain without actually being on it would help address this.
> > That, however, would be a partial solution. We'd still have to open the devtools network tab to see what domains were hit in the flow. Not a bit deal for those of us posting here. However, a regular user would need some other solution.
> 
> I understand what people are wanting to do, what I'm not understanding is why you'd go back to Chrome. I'm wondering if people just don't realize that Brave also has all the same settings that Chrome does (in brave://settings/content) or if there's some other reason. Maybe they don't want to deal with abp settings plus Brave settings? Seems like throwing the baby out with the bathwater because Brave provides many other protections independently of shields, but I guess that's a personal choice.

@bridiver It's the Brave Shield settings, including on redirects where a user can't simply toggle something in the shield UI.

Users just want ways (ideally programmatically) to be able to adjust per-domain and subdomain impacts of shields.

It's not that anyone is looking for more protection, just that we need to manage it in a more granular way.

There are 50+ comments of users here saying that this pain point specifically blocks their broader adoption of Brave.

These comments above outline some gaps in Brave's whitelisting syntax, both inability to subscribe to a third-party list, and not parsing certain rule syntax:

https://github.com/brave/brave-browser/issues/5290#issuecomment-572755768
https://github.com/brave/brave-browser/issues/5290#issuecomment-573074690

This can be solved by having the Brave Shields adopt mostly industry standards of:

- Programmatic prompting to add a rule
- Subscribing to blocklist rules
- Standard block/whitelist syntax

This is for the (brave://adblock/) end.

Not for anything to do with (brave://settings/content)

The default restrictive cases will not work for all user edge cases, we just want ways to easily manage this. Especially in corporate environments where an IT manager is trying to sync the whitelisting of multiple devices to work in an internal corporate environment.

**Comment by ashutoshsaboo** [comment]
One usecase where Brave is an immense pain related to this issue, is due to not being able to easily whitelist 302 domains, see details here: https://github.com/brave/brave-browser/issues/12782 . It's almost really difficult to block shields for such redirects, where if you couldn't block shields the website simply wouldn't open up. For internal websites locked behind auth, this is pretty standard usecase. In many of such cases atleast (where it's difficult to even inspect the correct website that needs to be whitelisted as they are varying urls in many cases), Brave is literally not usable at all and only a switch to Chrome/Firefox with ad blocking/tracking blocking addons is the only real alternative. 

I'm actually surprised adding such a toggle page and UI for it (https://github.com/brave/brave-browser/issues/12782) hasn't been prioritised yet. Clearly Brave already has support for such toggles, all they have to do is add some extra rules for whitelisting subdomains as well, and a simple UI to toggle such rules conveniently (https://github.com/brave/brave-browser/issues/12782). 50+ comments above clearly detailing about why is it a problem, with many evidences. 

Hope this issue and https://github.com/brave/brave-browser/issues/12782 gets prioritised soon.

**Comment by bridiver** [comment]
> These comments above outline some gaps in Brave's whitelisting syntax, both inability to subscribe to a third-party list, and not parsing certain rule syntax:
You can subscribe to third-party lists 
<img width="673" alt="image" src="https://user-images.githubusercontent.com/34129/183684402-741bae80-5287-4cfb-aa67-067fe75ba95e.png">

> This can be solved by having the Brave Shields adopt mostly industry standards of:
> 
> * Programmatic prompting to add a rule
> * Subscribing to blocklist rules
> * Standard block/whitelist syntax
> 
> This is for the (brave://adblock/) end.
@antonok-edm can comment on this, but it's a separate issue from content settings which is the subject of this issue.

I think you're still misunderstanding what I'm saying. I understand what people want to do in terms of settings, but moving back to Chrome does not give you any additional capabilities to manage settings and you lose settings that don't even exist in Chrome (like shields). Brave shields manages content settings (brave://settings/content). Brave-specific content settings like shields are not currently in brave://settings/content (we are discussing adding them), but with Chrome you lose them entirely. You can run Chrome with ABP or similar, but you could also run ABP on Brave and not give up everything else that Brave provides and that includes things that can't be replicated by an extension because they require changes to native code. Also if you're running ABP in Chrome you have less than a year to keep enjoying the network blocking currently provided by it https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/
Brave will continue to support manifest v2 until June 2023 when Chrome removes all support for the underlying apis it requires.

**Comment by OpenOrg-gg** [comment]
> > These comments above outline some gaps in Brave's whitelisting syntax, both inability to subscribe to a third-party list, and not parsing certain rule syntax:
> > You can subscribe to third-party lists
> 
> <img alt="image" width="673" src="https://user-images.githubusercontent.com/34129/183684402-741bae80-5287-4cfb-aa67-067fe75ba95e.png">
> 
> > This can be solved by having the Brave Shields adopt mostly industry standards of:
> > 
> > * Programmatic prompting to add a rule
> > * Subscribing to blocklist rules
> > * Standard block/whitelist syntax
> > 
> > This is for the (brave://adblock/) end.
> > @antonok-edm can comment on this, but it's a separate issue from content settings which is the subject of this issue.
> 
> I think you're still misunderstanding what I'm saying. I understand what people want to do in terms of settings, but moving back to Chrome does not give you any additional capabilities to manage settings and you lose settings that don't even exist in Chrome (like shields). Brave shields manages content settings (brave://settings/content). Brave-specific content settings like shields are not currently in brave://settings/content (we are discussing adding them), but with Chrome you lose them entirely. You can run Chrome with ABP or similar, but you could also run ABP on Brave and not give up everything else that Brave provides and that includes things that can't be replicated by an extension because they require changes to native code. Also if you're running ABP in Chrome you have less than a year to keep enjoying the network blocking currently provided by it https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/ Brave will continue to support manifest v2 until June 2023 when Chrome removes all support for the underlying apis it requires.

Yes, but ABP on Chrome honors the full whitelisting syntax that Brave doesn't seem to parse.

People are likely happy to give up the additional protections if means getting this to work as its a big painpoint, so they are switching.

The list subscribe is newer I hadn't noticed it but that is good. But there is still challenges in Brave not taking the full logic syntax in whitelisting.

**Comment by bridiver** [comment]
> Yes, but ABP on Chrome honors the full whitelisting syntax that Brave doesn't seem to parse.

ABP also runs on Brave with no reduction in functionality, but I think I actually see the issue now (and reason for going back to Chrome) because I was going to suggest that you could disable shields and manage everything through ABP on Brave (and not give up the other protections provided through native code), but there is no global setting to disable shields so there's no way to get to a "blank slate"

**Comment by bridiver** [comment]
I guess in the short term you could accomplish the same thing by defaulting all the individual settings to disabled. I realize this is not ideal and once again I'm not arguing against additional controls to provide the missing functionality in shields, I was just missing the issue of why you wouldn't just use Brave with ABP instead of Chrome, but now I understand.

**Comment by OpenOrg-gg** [comment]
> I guess in the short term you could accomplish the same thing by defaulting all the individual settings to disabled. I realize this is not ideal and once again I'm not arguing against additional controls to provide the missing functionality in shields, I was just missing the issue of why you wouldn't just use Brave with ABP instead of Chrome, but now I understand.

Glad we are on the same page. It can be hard to justify a switch, especially in a corporate environment if the solution is just to turn off the shields.

Shields are one of Brave's biggest benefits, so turning them off by default is a bit self defeating. 

The ability to programmatically manage the whitelist or whitelist subscriptions, and have that granular control would get many over this adoption hurdle. Hope we get to see this soon as there is a large push for corporate privacy and security these days :)

**Comment by bridiver** [comment]
> The ability to programmatically manage the whitelist or whitelist subscriptions, and have that granular control would get many over this adoption hurdle. Hope we get to see this soon as there is a large push for corporate privacy and security these days :)

I do want to make sure we keep the issue of managing content settings and adblock subscriptions/settings separate because they are two completely independent concerns. I believe there is already an open issue for adblock rules, but not sure.

**Comment by bsclifton** [comment]
Hi folks

Lots of comments here! Definitely an opportunity for us to do better. There are a lot of comments here - some getting into adblock syntax. But I want to make sure I understand the problem completely

It seems to me the major problem is the lack of an ability to provide Brave with a list of domains (or domain wildcards) and then assign a shield setting to that. For example, being able to apply a policy of shields down 1) on your companies intranet 2) sites that IT has already deemed as safe; ex: logmein.com

We have the same group policy that Chrome/Chromium does, but we don't have any policies around shields

Does that sound like an accurate characterization of the problem? The pain point is when users need to go and change shield settings on every single domain / subdomain by hand when there is a clear pattern?

**Comment by FearNaBoinne** [comment]
Sounds pretty accurate. Being able to say *Any subdomain of XYZ.COM" would go a long way. 

So if I can either add domains manually, or select (one of the) parent domain name(s), and a toggle "any subdomain", that would help.

So I'm on "login.api.somedomain.org" and I click on "somedomain", and I set shield options for that, have it apply to any subdomains of somedomain.org...

**Comment by ashutoshsaboo** [comment]
@bsclifton To add on to what you summarized which is mostly exhaustive, just please ensure there's also a proper UI to toggle for such settings, and not a regex notepad kinda settings toggle where one can add entries like gitignore/similar. You could do the latter as a v1 solution, but please also ensure there's a neat way also in UI to toggle such settings for the average user who wouldn't understand specifics of how to add configs in the editor manually (browsers are not just meant for power users after all :) ). See https://github.com/brave/brave-browser/issues/12782 for more details regarding UI request.

**Comment by bsclifton** [comment]
OK folks - going to close this one as a DUPLICATE; huge thanks to @ashutoshsaboo for helping characterize the issue and to @bridiver @spylogsster and others for helping me triage on our side. This is definitely an issue we want to fix and will be fixing

Please subscribe to https://github.com/brave/brave-browser/issues/12782 for updates 😄 The idea here is to add these Brave specific shield settings to brave://settings/content. These screens allow for wildcarding and manual input AND they respond to group policy. We may need an additional GitHub issue to ensure we process new group policy entries - but I'll create that separately

Thanks for your patience folks! 😄

**Comment by marijnbent** [comment]
Do I need to disable the shield on all my .test development sites? It blocks the Vite generated CSS on a localhost. 

<img width="598" alt="Image" src="https://github.com/user-attachments/assets/9e25d2b3-dbe8-4ff9-8de0-999995019d8a" />

<img width="374" alt="Image" src="https://github.com/user-attachments/assets/2a441f99-8169-40a1-864d-54deae556569" />
